ui/src/RecentFilesDrawer/AddTagToWorkflowPopover.tsx

```diff
@@ -34,22 +37,25 @@ export default function AddTagToWorkflowPopover({ workflow }: Props) {
   const [selectedTags, setSelectedTags] =
     useState<MultiValue<{ value: string; label: string }>>(initialTags);
 
-  useEffect(() => {
-    const loadTags = async () => {
-      const tags = await tagsTable?.listAll();
-      setAllTags(tags ?? []);
-    };
-    loadTags();
-  }, []);
+  const addTag = async () => {
+    const tagNameList =
+      (await tagsTable?.listAll().then((list) => list.map((f) => f.name))) ??
+      [];
+    if (tagNameList.includes(newTagName)) {
+      console.log(1111);
+      setSubmitError(true);
+    } else {
+      await tagsTable?.create(newTagName);
```

Comment:
i think we can just change this to put() and we don't need to check for duplicate

Reply:
done